src/cbor_decode.c

```diff
@@ -521,6 +520,9 @@ bool any_decode(cbor_state_t *state, void *result)
 	switch (major_type) {
 		case CBOR_MAJOR_TYPE_BSTR:
 		case CBOR_MAJOR_TYPE_TSTR:
+			if (value > (state->payload_end - state->payload)) {
```

Comment:
I don't get it. If the VALUE of `value` is bigger than the total length of the payload we should fail?

Reply:
Then it makes sense, for future me it would be good if that was described in a comment.
```suggestion
                        /* 'value' is the length of the BSTR or TSTR */
			if (value > (state->payload_end - state->payload)) {
```